### DISCUSSION TITLE
Github API (v3) 'list events for a repository' call only returns events for the last month ...

### DISCUSSION THREAD

**Discussion by hub-jens** [community/community#21952] [New to GitHub]
<p>Hi,</p>
<p>I’m trying to retrieve all events (comments, state/label changes) etc. from a repository. I’m using the ‘list events for a repository call’ (<a href="https://developer.github.com/v3/issues/events/#list-events-for-a-repository" rel="nofollow noopener">https://developer.github.com/v3/issues/events/#list-events-for-a-repository). </a>. )</p>
<p>I was surprised to see only a relatively low number of items returned.</p>
<p>Upon closer expection, I notices that the first event returned is the most recent, and the last event returned is exactly one month old. My repository goes back several years, so clearly there is a cut-off here.</p>
<p>My problem is that I cannot see that this is documented, nor any parameter that could control it (allowing me to get everything).</p>
<p>What is going on?</p>
<p>Regards,</p>
<p>Jens</p>

**Comment by hub-jens** [comment]
<p>Note, that I did try as well the get events call for a specific issue (<a href="https://developer.github.com/v3/issues/events/#list-events-for-an-issue" rel="nofollow noopener">https://developer.github.com/v3/issues/events/#list-events-for-an-issue</a>) . In this case, I do get all events, even going back several years. However, I would like to avoid doing a double loop (issues, then events) for get all repo events.</p>

**Comment by hub-jens** [comment, marked as answer]
<p>My mistake!</p>
<p>I did the wrong call to get events.</p>
<p>I used URL: …&lt;owner&gt;/&lt;repo&gt;/events</p>
<p>instead of </p>
<p>&lt;owner&gt;/&lt;repo&gt;/issues/events</p>
<p>Works fine.</p>
<p>(I got 22000+ events :slight_smile:</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/hub-jens">@hub-jens</a>,</p>
<p>So glad you were able to find the answer; thank you for coming back and updating this thread so that others might learn from your experience in the future!</p>
<p>I wanted to let you now that this post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information.</p>
<p>As you’ll notice, your Topic is now in the How to use Git and GitHub board. No action is needed on your part.</p>
<p>Let me know if you have any questions.</p>
<p>Cheers!</p>
